### PR TITLE
[New Rule] GCP Logging Sink Deletion

### LINKED ISSUE #243: [New Rule] GCP Logging Sink Deletion
## Description
Identifies a Logging sink deletion in Google Cloud Platform (GCP). Every time a log entry arrives in a project, folder, billing account, or organization resource, Logging compares the log entry to the sinks in that resource. Each sink whose filter matches the log entry writes a copy of the log entry to the sink's export destination. An adversary may delete a Logging sink to evade detections.

## Required Info

- **Eventing Sources:**
`filebeat-*`

- **Platforms**
`GCP`

- **Target ECS Version:** `1.6.0`
- **New fields required in ECS for this?**
- **Related issues or PRs**

## Optional Info
- **References:**
https://cloud.google.com/logging/docs/export

## Syntax
```
event.module:googlecloud and event.dataset:googlecloud.audit and event.action:google.logging.v*.ConfigServiceV*.DeleteSink and event.outcome:success
```

## Example Data
image

### PR DESCRIPTION
**Pull request by bm11100**
## Issue(s)
resolves #243 

## Description

Identifies a Logging sink deletion in Google Cloud Platform (GCP). Every time a log entry arrives, Logging compares the log entry to the sinks in that resource. Each sink whose filter matches the log entry writes a copy of the log entry to the sink's export destination. An adversary may delete a Logging sink to evade detection.

## Checklists
Use ~~strikethroughs~~ to remove items which are not applicable to this issue.

### For submitter
- [x] Verify all query fields co-exist in a single event source (*beats) or annotated otherwise
  -- Query field process.name.text doesn't currently exist in endgame-* eventing data, but does exist for winlogbeats, and is case insensitive.
- [x] Verify all query fields co-exist within a single ECS version and annotated the minimum version
  -- 1.4.0
- [x] Verified the query detects the intended event(s)
  - detonate
  - search in _discover_
  - trigger as a custom signal
- [x] Create rule using `create-rule`
- [x] Verify and convert to standard linting (`toml-lint -f <rule-file>`)
- [x] Run tests using `pytest -x -v unit_tests` or `make test`
- [ ] Internal search to determine noise and FP

### For reviewers
- [ ] Verify existing rule for activity doesn't exist as a siem or endpoint rule (unless intentionally specified in issue)
- [ ] Verify all query fields co-exist in a single event source (*beats) or annotated otherwise
- [ ] Verify all query fields co-exist within a single ECS version and annotated the minimum version
- [ ] Internal search to determine noise and FP
- [ ] Verify metadata accuracy and spelling

### CODE REVIEW

**Comment by bm11100** [comment]
Note for reviewers: The default run and look back time appear to be sufficient as the GCP data, unlike AWS and Azure, doesn't batch send every 10-15 min. I also didn't see an applicable Mitre tactic/technique for this open to suggestions though if we want to add one.

**Review by threat-punter — APPROVED** [review]
LGTM. Should the file name be prefixed with "defense_evasion"? Even though there is not a suitable technique ID to map this rule to, I feel like it fits under the Defense Evasion tactic at least.

**Comment by bm11100** [comment]
> LGTM. Should the file name be prefixed with "defense_evasion"? Even though there is not a suitable technique ID to map this rule to, I feel like it fits under the Defense Evasion tactic at least.

Added the `Impair Defenses` technique

File: rules/gcp/defense_evasion_gcp_logging_sink_deletion.toml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+query = '''
+event.module:googlecloud and event.dataset:googlecloud.audit and event.action:google.logging.v*.ConfigServiceV*.DeleteSink and event.outcome:success
+'''
+
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1562"
```
Comment (brokensound77) [inline rules/gcp/defense_evasion_gcp_logging_sink_deletion.toml:41]: 🎈

**Review by brokensound77 — APPROVED** [review]
LGTM 👍

**Pull request merged**